hbase_sql/src/main/java/com/splicemachine/stream/ResultStreamer.java

```diff
@@ -75,7 +75,7 @@ public ResultStreamer(OperationContext<?> context, UUID uuid, String host, int p
         this.numPartitions = numPartitions;
         this.batches = batches;
         this.batchSize = batchSize;
-        this.permits = new Semaphore(batches - 1); // we start with one permit taken
+        this.permits = new Semaphore(0); // we start with one permit taken
```

Comment:
With this change we have to wait until the client gives us the go ahead to start sending data from the server, right? It adds a little bit of delay, maybe not a big deal but it could hurt us when dealing with those very big fetch jobs with thousands of partitions to fetch. In any case the comment has to be fixed.